mikeio1d/pandas_extension/result_frame_aggregator.py

```diff
@@ -0,0 +1,332 @@
+from __future__ import annotations
+
+from typing import TYPE_CHECKING
+
+from typing import Callable
+
+if TYPE_CHECKING:
+    from typing import Any
+    from typing import List
+    from typing import Dict
+
+from dataclasses import fields
+
+import pandas as pd
+
+from mikeio1d.pandas_extension import groupby_level
+from mikeio1d.pandas_extension import compact_dataframe
+from mikeio1d.various import make_list_if_not_iterable
+from mikeio1d.quantities import TimeSeriesId
+
+
+class ResultFrameAggregator:
+    """
+    Aggregates a MIKE IO 1D result DataFrame into scalar values associate with entities.
+
+    Parameters
+    ----------
+    agg : str or callable
```

Comment:
By levels we mean, pandas multiindex levels?
I also think it won't hurt to write an example for argument, though it also is present below. **For example, max of values, etc.**